### PR TITLE
Add flow for recovering access to account if Redux store is wiped

### LINKED ISSUE #189: Add flow for recovering access to account if Redux store is wiped
### Expected Behavior

If the Redux store is wiped, users can recover access to their account.

### Current Behavior

Users have no way of recovering access other than reimporting their Account Key.

### PR DESCRIPTION
**Pull request by gnardini**
### Description

When we detect that there is no account stored on the Redux store but there is one on the geth wallet, it means that the Redux store was unexpectedly wiped. To recover from this situation, this PR adds a new screen notifying the user that there's been a user and letting them know that they'll need to input their name and phone number again.

We'll be able to skip the name step in the future when CIP-8 lands.

https://user-images.githubusercontent.com/6062888/113381557-a6826a00-9355-11eb-8291-3d983c9b5b6f.mov

### Other changes

- Added an option to the hidden dev settings to erase the Redux store in order to be able to test this.

### Tested

Manually using the hidden dev setting.

### Related issues

- Fixes #189

### Backwards compatibility

N/A

### CODE REVIEW

File: packages/mobile/src/utils/accountChecker.ts [hunk 1]
```diff
@@ -13,9 +15,10 @@ export function* checkAccountExistenceSaga() {
   const gethAccounts: string[] = yield wallet.getAccounts()
   const reduxAddress: string = yield select(currentAccountSelector)
   if (!reduxAddress && gethAccounts.length > 0) {
-    // TODO: Try to recover access to the account.
+    const account = gethAccounts[0]
```
Comment (jeanregisser) [inline packages/mobile/src/utils/accountChecker.ts:18]: Is it guaranteed that the right account we want will always be the first one?

File: packages/mobile/src/import/ImportWallet.tsx [hunk 2]
```diff
@@ -90,12 +96,28 @@ export class ImportWallet extends React.Component<Props, State> {
...
+    }
   }
 
   componentWillUnmount() {
     this.props.navigation.removeListener('focus', this.checkCleanBackupPhrase)
   }
 
+  async autocompleteSavedMnemonic() {
+    const wallet = await getWalletAsync()
+    const account = wallet?.getAccounts()[0]
```
Comment (jeanregisser) [inline packages/mobile/src/import/ImportWallet.tsx:113]: Though unlikely, there's a chance that this account value could differ from the one passed when initially opening `Screens.StoreWipeRecoveryScreen`, if we make some changes to this in the future.

File: packages/mobile/src/account/actions.ts [hunk 3]
```diff
@@ -149,6 +154,7 @@ export interface SetRetryVerificationWithFornoAction {
 export interface ClearStoredAccountAction {
   type: Actions.CLEAR_STORED_ACCOUNT
   account: string
+  softDelete: boolean
```
Comment (jeanregisser) [inline packages/mobile/src/account/actions.ts:157]: Making the naming more explicit?
```suggestion
  onlyReduxState: boolean
```

File: packages/mobile/src/account/saga.ts [hunk 4]
```diff
@@ -44,19 +44,21 @@ export function* setPincode({ pincodeType }: SetPincodeAction) {
...
 }
 
-function* clearStoredAccountSaga({ account }: ClearStoredAccountAction) {
+function* clearStoredAccountSaga({ account, softDelete }: ClearStoredAccountAction) {
   try {
-    yield call(removeAccountLocally, account)
-    yield call(clearStoredMnemonic)
-    yield call(ValoraAnalytics.reset)
-    yield call(deleteNodeData)
+    if (!softDelete) {
```
Comment (jeanregisser) [inline packages/mobile/src/account/saga.ts:49]: nit:
```suggestion
    if (!softDelete) { return }
```

**Review by jeanregisser — COMMENTED** [review]
Awesome work! 🚀 
Left a couple of small comments.

File: packages/mobile/src/utils/accountChecker.ts (hunk 1 above)
Comment (gnardini) [reply packages/mobile/src/utils/accountChecker.ts:18]: I'm not sure really, but I don't know why we would have another? In the tests I've made there's only one and I don't know of a way to have more than one, though I guess it's possible. What would you expect the behavior to be if there's more than one?

File: packages/mobile/src/account/saga.ts (hunk 4 above)
Comment (gnardini) [reply packages/mobile/src/account/saga.ts:49]: We do want the last two lines with the `persistor.flush` and `restartApp` to run though.

File: packages/mobile/src/utils/accountChecker.ts (hunk 1 above)
Comment (jeanregisser) [reply packages/mobile/src/utils/accountChecker.ts:18]: yes not too sure either, let's go with this for now.

File: packages/mobile/src/account/saga.ts (hunk 4 above)
Comment (jeanregisser) [reply packages/mobile/src/account/saga.ts]: oh sorry didn't see that 👍

**Review by jeanregisser — APPROVED** [review]
w00t! 🚢 👍

**Comment by Celoqa** [comment]
Hi @gnardini Currently we are  using latest Test flight 1.12.2(52) build in Mainnet server hence we are not able see to any Reduxe store option on settings menu.
**Note:** We are able to se Alfajores build 1.12.1(50)
**Need clarifications:** Is there any timestamp to automatically logout when user keep app in background For 1-2 days.   
Can you please do the needfull.

**Comment by gnardini** [comment]
@Celoqa By tapping on the title of the Settings screen 5 times you can see the dev options. They show up on the bottom of the Settings screen.

**Comment by Celoqa** [comment]
Thanks @gnardini  for the steps , now we are able to see "Wipe Redux store" option from setting menu after tapping on "Settings" text on settings page for 5 times.

**Comment by Celoqa** [comment]
@gnardini  We have verified this using latest Test flight build 1.12.2(50) &  Android internal build V1.12.2(1004294345) and observed the followings.

- "We need you to setup your account" text with Restore Account button is shown when user open the app after automatically logged out.
- User redirects to Restore my account page after tapping on Restore account button 
- Account key get pasted automatically & user should able to restore the account.
- "Balance $0.00" popup is shown is user as $0.00 in account after automatically pasted account key.
- User redirects to Home page after restoring account & able to see all activity.

**Observed issue:** "Oops! Something went wrong" error message is shown after restoring the account.
**Frequency:** 40%
**Device:** iPhone XR (14.4)
**Attachment:** [SomethingWentWrongErrorMessage.png](https://drive.google.com/file/d/1Th_FKWDS7eZ0qYDP9B8DMy_8MyxAvfPH/view?usp=sharing)

**Comment by gnardini** [comment]
Hi @Celoqa the latest updated (54 on Testflight and 1004294347 on Android) has a fix for this. Could you test this again please?

Thanks!

**Comment by Celoqa** [comment]
Sure @gnardini We will verify on latest builds Android internal build V1.12.2(1004294347) & Test flight build V1.12.2(54) and will update you by EOD.
Thanks !!

**Comment by Celoqa** [comment]
@gnardini We have verified this again using latest Test flight build 1.12.2(54) & Android internal build V1.12.2(1004294347) and observed the followings.

- "We need you to setup your account" text with Restore Account button is shown when user open the app after automatically logged out.
- User redirects to Enter pin page after tapping on restore button 
- User able to redirect to restore page by entering first name on account info page 
- Account key get pasted automatically & user should able to restore the account.
- "Balance $0.00" popup is shown is user as $0.00 in account after automatically pasted account key.
- User redirects to Home page after restoring account & able to see all activity.

**Note:** "Oops! something went wrong error message is not show & user is able to restore the account without any issues.

**Comment by Celoqa** [comment]
https://valora.testquality.com defect [D4](https://valora.testquality.com/project/12128/result/2397005/defect/7771). Do not remove or edit.

**Pull request merged**